tuxedo-core/src/constraint_checker.rs

```diff
@@ -34,32 +31,43 @@ pub trait SimpleConstraintChecker: Debug + Encode + Decode + Clone {
 /// A single constraint checker that a transaction can choose to call. Checks whether the input
 /// and output data from a transaction meets the codified constraints.
 ///
-/// This full ConstraintChecker should only be used if there is more that a piece wants to do such
-/// as check the verifier information in some unique way.
+/// This full ConstraintChecker should only be implemented if the piece logic cannot be expressed with
+/// the SimpleConstraintChecker. For example, if you need to enforce the verifier is a particular type
+/// or contains a certain value. Another reason would be if you need to implement an inherent.
```

Comment:
I was thinking maybe we could add an `InherentConstraintChecker` trait instead.

Reply:
Yeah. This is a good idea. It would simplify a lot of things. This is roughly what I had in mind when I put  `(Deferred to followup) Rework relationship between constraint checkers and inherent hooks.` as one of the checklist items in the PR description.